Cannot use the Nix C API from another project

## Describe the bug

The cflags property in the generated `nix-*-c` pkgconfig files causes issues with external C projects. 
```
[1/2] Compiling C object nixc.p/main.c.o
FAILED: nixc.p/main.c.o 
clang -Inixc.p -I. -I.. -I/usr/include -I/usr/include/libcpuid -flto -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Wpedantic -std=c2x -O2 -g -std=c++2a -std=c++2a -DBLAKE3_DLL -DBLAKE3_USE_TBB -MD -MQ nixc.p/main.c.o -MF nixc.p/main.c.o.d -o nixc.p/main.c.o -c ../main.c
error: invalid argument '-std=c++2a' not allowed with 'C'
ninja: build stopped: subcommand failed.
```

The issue is that the pc files contain c++ specific arguments. I'm not clear on the what the proper fix would be in Nix's meson.build setup yet:
`Cflags: -I${includedir} -std=c++2a`

Edit: probably related: https://github.com/NixOS/nix/blob/b4bea57667eb5413dbc808dd11c10106ef91c770/nix-meson-build-support/export/meson.build#L19

## Steps To Reproduce

- Install the Nix c api libraries and headers, e.g. on arch `pacman -S nix` 
- Create a simple meson.build file as seen below along with a demo c file
- Build it with clang (or gcc?)
```
project(
    'nix-demo',
    'c',
    default_options: [
        'c_std=c2x',
        'warning_level=3',
        'buildtype=debugoptimized',
        'b_lto=true',
    ],
)

nix_expr = dependency('nix-expr-c')
nix_util = dependency('nix-util-c')

executable('nix-demo', 'main.c', dependencies: [nix_expr, nix_util])
```

## Expected behavior

There should not be a -std=c++2a flag being passed down to external projects that rely on the pkgconfig files of Nix.

## Metadata

<!-- Please insert the output of running `nix-env --version` below this line -->
nix-env (Nix) 2.28.3

## Additional context

This occurs on Arch Linux but I'd assume the same issue would happen with nixpkgs/NixOS.

## Checklist

<!-- make sure this issue is not redundant or obsolete -->

- [x] checked [latest Nix manual] \([source])
- [x] checked [open bug issues and pull requests] for possible duplicates

[latest Nix manual]: https://nixos.org/manual/nix/unstable/
[source]: https://github.com/NixOS/nix/tree/master/doc/manual/source
[open bug issues and pull requests]: https://github.com/NixOS/nix/labels/bug

---

Add :+1: to [issues you find important](https://github.com/NixOS/nix/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).

--- a/nix-meson-build-support/export/meson.build
+++ b/nix-meson-build-support/export/meson.build
@@ -11,12 +11,18 @@ endforeach
 requires_public += deps_public
 
 extra_pkg_config_variables = get_variable('extra_pkg_config_variables', {})
+
+extra_cflags = []
+if not meson.project_name().endswith('-c')
+    extra_cflags += ['-std=c++2a']
+endif
+
 import('pkgconfig').generate(
   this_library,
   filebase : meson.project_name(),
   name : 'Nix',
   description : 'Nix Package Manager',
-  extra_cflags : ['-std=c++2a'],
+  extra_cflags : extra_cflags,
   requires : requires_public,
   requires_private : requires_private,
   libraries_private : libraries_private,

